tools/data-model-exporter/data_model_exporter/dmExporter.py

```diff
@@ -0,0 +1,164 @@
+                else:
+                    classCount += 1
+                    classList.append(line.strip())
+        print (str(classCount)+" classes parsed from "+args.classPath)
+        print ("classList:  "+str(classList))
+        return args.filePath, classList
+    else:
+        print ("Error parsing arguments, please try again...")
+
+
+def run(file_path, class_name):
+    with open(file_path, 'r') as ttl_file:
+        rdf_term = Terra.term(class_name)
+        # parse the file
+        g = Graph()
+        g.parse(ttl_file, format='turtle')
+        properties = {
+            "describedBy": {
+                "description": "The URL reference to the JSON Schema that defines this object.",
+                "type": "string"
+            },
+            "id": {
+                "description": "UUID for this entity.",
+                "type": "string"
+            }
+        }
+        # 'required' properties are those with a cardinality of exactly 1, and the hardcoded 'id' property
+        required = ['id','describedBy']
+
+        ## TODO punting on rdfs:subClassOf linkage
+
+        # pull all RDF triples with the given rdf_term as the 'subject'
+        term_triples = g.triples((rdf_term, None, None))
+        for triple in term_triples:
+            # any OWL equivalentClass predicate is a 'property'
+            if triple[1] == OWL.equivalentClass:
+                # the 'object' node reached via the equivalentClass predicate is a 'blank node' in
+                # RDF; this node is anonymous and serves as a container for composite property information
+                container_node = triple[2]
+
+                # required prop?
+                cardinality = g.value(container_node, OWL.cardinality, None)
+                prop = g.value(container_node, OWL.onProperty, None)
+                # subject = container_node, predicate = OWL.onProperty, object)
+                # TODO fix up the logic we are using on rdfs:range vs other
+                ref = prop.n3(g.namespace_manager)
+
+                # second traversal, using objects from 1st as the subject
+                # processing subclasses with rdfs:range
+                # have an RDFS range. We should be pulling all of the rdfs:range
+                # bc if an object exists with rdfs:range, we will put it
+
+                # pull the rdfsRangeValue directly out using Graph.value function
+                # this results in nulls for dct: terms
+                rdfsRangeValue = g.value(prop, RDFS.range)
+
+                # using a try/catch for when we run into UnboundLocalError: 'rdfsRangeValue' referenced before assignment
+                try:
+                    properties[prop.n3(g.namespace_manager)] = {
+                        'description': ref,
+                        # TODO: pull the URL reference out of the graph
+                        '$ref': rdfsRangeValue,
+                    }
+                except Exception as e:
+                    raise e
```

Comment:
This is an odd construct -- we're catching the exception only to log and raise again. Is this actually necessary?